test/Microsoft.IdentityModel.Validators.Tests/MicrosoftIdentityIssuerValidatorTest.cs

```diff
@@ -305,6 +306,31 @@ public void Validate_IssuerNotInTokenValidationParameters_ReturnsIssuer(string t
             TestUtilities.AssertFailIfErrors(context);
         }
 
+        [Theory]
+        [InlineData(ValidatorConstants.ClaimNameTid, ValidatorConstants.AadIssuer)]
+        [InlineData(ValidatorConstants.TenantId, ValidatorConstants.AadIssuer)]
+        [InlineData(ValidatorConstants.ClaimNameTid, ValidatorConstants.V1Issuer)]
+        [InlineData(ValidatorConstants.TenantId, ValidatorConstants.V1Issuer)]
+        public void ValidateJsonWebToken_ReturnsIssuer(string tidClaimType, string issuer)
+        {
+            var context = new CompareContext();
+            var validator = new AadIssuerValidator(_httpClient, issuer);
+            var tidClaim = new Claim(tidClaimType, ValidatorConstants.TenantIdAsGuid);
+
+            var issClaim = new Claim(ValidatorConstants.ClaimNameIss, issuer);
+            List<Claim> claims = new List<Claim>();
+            claims.Add(tidClaim);
+            claims.Add(issClaim);
+
+            var jsonWebToken = new JsonWebToken(Default.Jwt(Default.SecurityTokenDescriptor(Default.SymmetricSigningCredentials, claims)));
```

Comment:
Can we use one of the JsonWebTokenHandler.CreateToken (perhaps the one that takes SecurityTokenDescriptor) methods instead and avoid adding SecurityTokenDescriptor(credentials, claims)?

Reply:
I tried to do that, but I don't want an encrypted token and also needed some custom values: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1801/files#diff-20c5742d81614f58ad5fd9e5ccc34931d030cbaf5b182bfbc8607707aa3a275bR811 and the claims were not getting added. thoughts?

Reply:
@jennyf19 using SecurityTokenDescriptor does not imply a JWE. Only if the EncryptingCredentials are set will a JWE be created.
You can use this property on SecurityTokenDescriptor to add additional payload claims: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/669515a4ab1e3201768e0d74cd5097db0ba4152b/src/Microsoft.IdentityModel.Tokens/SecurityTokenDescriptor.cs#L88

Reply:
Add this property to add additional header claims: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/669515a4ab1e3201768e0d74cd5097db0ba4152b/src/Microsoft.IdentityModel.Tokens/SecurityTokenDescriptor.cs#L97